src/main/java/org/assertj/core/api/Assumptions.java

```diff
@@ -0,0 +1,757 @@
+   *
+   * @param actual the actual value.
+   * @return the created assumption for assertion object.
+   */
+  @CheckReturnValue
+  @SuppressWarnings("unchecked")
+  public static <ELEMENT> FactoryBasedNavigableIterableAssert<IterableAssert<ELEMENT>, Iterable<? extends ELEMENT>, ELEMENT, ObjectAssert<ELEMENT>> assumeThat(Iterator<? extends ELEMENT> actual) {
+    return enhanceAsAssumption(IterableAssert.class, Iterator.class, actual);
+  }
+
+  /**
+   * Creates a new instance of <code>{@link ListAssert}</code> assumption.
+   *
+   * @param actual the actual value.
+   * @return the created assumption for assertion object.
+   */
+  @CheckReturnValue
+  @SuppressWarnings("unchecked")
+  public static <ELEMENT> FactoryBasedNavigableListAssert<ListAssert<ELEMENT>, List<? extends ELEMENT>, ELEMENT, ObjectAssert<ELEMENT>> assumeThat(List<? extends ELEMENT> actual) {
+    return enhanceAsAssumption(ListAssert.class, List.class, actual);
+  }
+
+  /**
+   * Creates a new instance of <code>{@link ObjectArrayAssert}</code> assumption.
+   *
+   * @param actual the actual value.
+   * @return the created assumption for assertion object.
+   */
+  @CheckReturnValue
+  @SuppressWarnings("unchecked")
+  public static <T> AbstractObjectArrayAssert<?, T> assumeThat(T[] actual) {
+    return enhanceAsAssumption(ObjectArrayAssert.class, Object[].class, actual);
+  }
+
+  /**
+   * Creates a new instance of <code>{@link MapAssert}</code> assumption.
+   *
+   * @param <K> the type of keys in the map.
+   * @param <V> the type of values in the map.
+   * @param actual the actual value.
+   * @return the created assumption for assertion object.
+   */
+  @CheckReturnValue
+  @SuppressWarnings("unchecked")
+  public static <K, V> AbstractMapAssert<?, ?, K, V> assumeThat(Map<K, V> actual) {
+    return enhanceAsAssumption(MapAssert.class, Map.class, actual);
+  }
+
+  /**
+   * Creates a new instance of <code>{@link ThrowableAssert}</code> assumption.
+   *
+   * @param actual the actual value.
+   * @return the created assumption for assertion object.
+   */
+  @CheckReturnValue
+  public static AbstractThrowableAssert<?, ? extends Throwable> assumeThat(Throwable actual) {
+    return enhanceAsAssumption(ThrowableAssert.class, Throwable.class, actual);
+  }
+
+  /**
+   * Allows to capture and then assume on a {@link Throwable} (done easier when used with Java 8 lambdas).
+   *
+   * @param shouldRaiseThrowable The {@link ThrowingCallable} or lambda with the code that should raise the throwable.
+   * @return the created assumption for assertion object.
+   */
+  @CheckReturnValue
+  public static AbstractThrowableAssert<?, ? extends Throwable> assumeThatThrownBy(ThrowingCallable shouldRaiseThrowable) {
+    return enhanceAsAssumption(ThrowableAssert.class, Throwable.class, catchThrowable(shouldRaiseThrowable));
+  }
+
+  @SuppressWarnings("unchecked")
+  private static <ASSERTION, ACTUAL> ASSERTION enhanceAsAssumption(Class<ASSERTION> assertionType,
```

Comment:
I'd like to rename `enhanceAsAssumption`, IMO it's not really an enhancement since we change the behavior of the called code, I was thinking of the simple `asAssumption` or `wrapAsAssumption`.